core/src/test/scala/unit/kafka/log/LogSegmentTest.scala

```diff
@@ -65,6 +68,29 @@ class LogSegmentTest {
     Utils.delete(logDir)
   }
 
+  /**
+   * If the maximum offset beyond index, appended to the log section, it throws LogSegmentOffsetOverflowException
```

Comment:
This comment is not clear enough (ex: what is maximum offset mean in this test?). Maybe:
```
LogSegmentOffsetOverflowException should be thrown while appending the logs if:
1. largestOffset - baseOffset < 0
2. largestOffset - baseOffset > Integer.MAX_VALUE
```

Reply:
Good idea. I've modified it